backend/src/main/java/com/empire/World.java

```diff
@@ -2064,6 +2047,66 @@ void eatFood() {
 		}
 
 		void growPopulations() {
+			// Emigration
+			class Emigration {
+				final Region source;
+				final Region sink;
+				final double population;
+				public Emigration(Region source, Region sink, double population) {
+					this.source = source;
+					this.sink = sink;
+					this.population = population;
+				}
+			}
+			List<Emigration> emigrations = new ArrayList<>();
+			for (Region r : regions) {
+				double eligibleToEmigrate = r.population * r.unrestPopular * Constants.emigrationFactor;
+				if (starvingRegions.contains(r)) eligibleToEmigrate *= (1 + Constants.emigrationStarvationMod);
+				eligibleToEmigrate = Math.min(eligibleToEmigrate, r.population - 1); // Emigrating the last person causes a divide by zero error.
+				List<Region> destinations = new ArrayList<Region>();
+				for (Region n : r.getNeighbors(World.this)) {
+					if (n.type == Region.Type.LAND) destinations.add(n);
+					else for (Region nn : n.getNeighbors(World.this)) {
+						if (nn != r && n.type == Region.Type.LAND) destinations.add(nn);
+					}
+				}
+				destinations.removeIf(d -> d.unrestPopular >= r.unrestPopular);
+				destinations.removeIf(d -> starvingRegions.contains(d));
+				destinations.removeIf(d -> !d.getKingdom().equals(r.getKingdom()) && (patrolledRegions.contains(r) || getNation(d.getKingdom()).getRelationship(r.getKingdom()).refugees == Relationship.Refugees.REFUSE));
+				if (destinations.isEmpty() || eligibleToEmigrate < 1) continue;
+				double totalWeight = 0;
+				for (Region d : destinations) totalWeight += d.calcImmigrationWeight();
+				for (Region d : destinations) {
+					emigrations.add(new Emigration(r, d, d.calcImmigrationWeight() / totalWeight * eligibleToEmigrate));
+				}
+			}
+			Map<Region, Double> netMotion = new HashMap<>();
+			for (Emigration e : emigrations) {
+				netMotion.merge(e.source, -e.population, Double::sum);
+				double unhappyCitizens = e.source.population * e.source.unrestPopular;
+				e.source.population -= e.population;
+				e.source.unrestPopular = (unhappyCitizens - e.population) / e.source.population;
+				netMotion.merge(e.sink, e.population, Double::sum);
+				unhappyCitizens = e.sink.population * e.sink.unrestPopular;
+				e.sink.population += e.population;
+				e.sink.unrestPopular = (unhappyCitizens + e.population) / e.sink.population;
+			}
+			for (String k : kingdoms.keySet()) {
```

Comment:
From the player's point of view, this could be really frustrating not knowing more of the particulars around where population is coming from and being lost if you are only notified of the net changes.  Without slicing it too finely numbers-wise, just having a hint of where positive/negative flows are coming from would probably suffice.

Reply:
Hm, you do get the details of which regions go up and down, I think the next step would be to tell you each of the emigrations - not sure if that's the too much detail you're talking about. Do you have an example of what you have in mind?

Reply:
I was thinking additional info along the lines of "You gained pop. from regions A, B, C. You lost pop. to regions X, Y, Z"

Reply:
Added more detail.